### PR TITLE
Wallet renewal

### LINKED ISSUE #671: Improved secret storage
1. If no wallet created yet, it is possible to create it by running the node with a certain flag. Password and, optionally, seed are to be provided during wallet generation (via command line), if no seed is provided it is generated from system randomness. Password is used to encrypt a wallet file (w. seed and wallet historical operations).

A wallet could be imported using either
* seed directly
* BIP39 mnemonic (if supported)

To password-protect the seed on the filesystem (in the wallet file), the encryption key should be derived from the password using a password hashing function (Argon2, PBKDF2, or scrypt), with sufficiently many iterations and a random salt. For example 50,000 PBKDF2 iterations.

If a wallet needs be generated then  system randomness should be fine (SecureRandom's). It may be better to generate a BIP39 mnemonic, because many interfaces and hardware wallets only support import via BIP39. And of course maybe the most important is to make sure there is a backup of the wallet/seed/mnemonic .

2.  If wallet is created, it is possible to open it, again, by providing a certain flag when starting a node.

### PR DESCRIPTION
**Pull request by oskin1**
Close #671

### CODE REVIEW

File: src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala [hunk 1]
```diff
@@ -64,14 +64,13 @@ class ErgoTransactionSpec extends ErgoPropertyTest {
     )
     val tx = ErgoTransaction(inputs: IndexedSeq[Input], outputCandidates: IndexedSeq[ErgoBoxCandidate])
 
-    val bytesToSign = "02c95c2ccf55e03cac6659f71ca4df832d28e2375569cec178dcb17f3e2e5f77420000ca796a4fc9c0d746a69702a77bd78b1a80a5ef5bf5713bbd95d93a4f23b27ead00000000028094ebdc030008cd0326df75ea615c18acc6bb4b517ac82795872f388d5d180aac90eaa84de750b942e8070000c0843d1005040004000e36100204cf0f08cd0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798ea02d192a39a8cc7a701730073011001020402d19683030193a38cc7b2a57300000193c2b2a57301007473027303830108cdeeac93b1a57304e8070000"
```
Comment (catena2w) [inline src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala]: Why is this changed?
Test vectors are here to be stable)

**Comment by oskin1** [comment]
> In addition to review notes, please update the `papers/yellow/img/box-transition.png` file.

This PR does not affect boxes transition, this will be done in #695

File: src/main/resources/application.conf [hunk 2]
```diff
@@ -150,11 +150,32 @@ ergo {
...
+
+        # Desired bit-length of the derived key (PBKDF2 param)
+        dkLen = 256
+      }
+
+    }
+
+    # Generating seed length in bits
+    # Options: 128, 160, 192, 224, 256
+    seedStrengthBits = 160
```
[the remark this answers is not in the document]
Comment (oskin1) [reply src/main/resources/application.conf]: Such options are recommended by BIP32.
